plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Simplify.hs

```diff
@@ -62,5 +66,11 @@ simplifyTerm opts = simplifyNTimes (_soMaxSimplifierIterations opts)
     simplifyStep _ =
         floatDelay
             >=> pure . forceDelayCancel
+            >=> pure . caseOfCase'
             >=> pure . caseReduce
             >=> inline (_soInlineHints opts)
+
+    caseOfCase' :: Term name uni fun a -> Term name uni fun a
+    caseOfCase' = case eqT @fun @DefaultFun of
+        Just Refl -> caseOfCase
+        Nothing   -> id
```

Comment:
If you are checking that I think the specific builtin is in the default universe.
```suggestion
    caseOfCase' = case (eqT @fun @DefaultFun, eqT @uni @DefaultUni) of
        (Just Refl, Just Refl) ->caseOfCase
        Nothing   -> id
```

Reply:
There's no need to check `eqT @uni @DefaultUni`. It needs to check `fun` because case-of-case only works on `DefaultFun`.